src/lib/scanner.ts

```diff
@@ -137,6 +139,30 @@ export class ScanRunner {
         });
     }
 
+    /**
+     * Uses the provided arguments to run a Salesforce Code Analyzer command.
+     * @param args The arguments to be supplied
+     */
+    private async invokeDfaAnalyzer(args: string[], context: vscode.ExtensionContext): Promise<ExecutionResult> {
```

Comment:
I'm prepared to have @stephen-carter-at-sf disagree with me about this, but since this method is almost entirely the same as the `invokeAnalyzer` method, I'd like it if we could either have the `context` be an optional extra parameter for that method, or have them both be wrappers around a third method with the same internal logic. It would cut down on duplicate code.

Reply:
I could go either way with this small amount of code. But if we wanted to share code, then one option is to  have the invokeAnalyzer take a function handle that processes the pid (where its default is a no-op for the standard runs). Then invokeDfaAnalyzer can call the invokeAnalyzer with a function handle that actually does the workspaceState update with the pid/undefined values.

Reply:
I would leave this as is mainly because this whole pid based execution would change to a unified way by calling core in a matter of months.